fitlins/interfaces/afni.py

```diff
@@ -0,0 +1,637 @@
+            the affine is considered the same for all.
+
+        runtime : nipype runtime object
+
+        nistats_flm : nistats.first_level_model.First_Level_Model
+
+        contrasts : Object returned by nistats.utils.contrasts.prepare_contrasts
+        events : pandas Dataframe or string or list of pandas DataFrames or
+                   strings
+
+            fMRI events used to build design matrices. One events object
+            expected per run_img. Ignored in case designs is not None.
+            If string, then a path to a csv file is expected.
+
+        confounds : pandas Dataframe or string or list of pandas DataFrames or
+                   strings
+
+            Each column in a DataFrame corresponds to a confound variable
+            to be included in the regression model of the respective run_img.
+            The number of rows must match the number of volumes in the
+            respective run_img. Ignored in case designs is not None.
+            If string, then a path to a csv file is expected.
+
+        design_matrices : pandas DataFrame or list of pandas DataFrames,
+            Design matrices that will be used to fit the GLM. If given it
+            takes precedence over events and confounds.
+
+        """
+        import sys
+        import time
+        from nilearn._utils.niimg_conversions import check_niimg
+
+        from nistats.design_matrix import make_first_level_design_matrix
+        from nistats.utils import (
+            _check_run_tables,
+            _check_events_file_uses_tab_separators,
+        )
+
+        # Check arguments
+        # Check imgs type
+        if events is not None:
+            _check_events_file_uses_tab_separators(events_files=events)
+        if not isinstance(run_imgs, (list, tuple)):
+            run_imgs = [run_imgs]
+        if design_matrices is None:
+            if events is None:
+                raise ValueError("events or design matrices must be provided")
+            if nistats_flm.t_r is None:
+                raise ValueError(
+                    "t_r not given to FirstLevelModel object"
+                    " to compute design from events"
+                )
+        else:
+            design_matrices = _check_run_tables(
+                run_imgs, design_matrices, "design_matrices"
+            )
+        # Check that number of events and confound files match number of runs
+        # Also check that events and confound files can be loaded as DataFrame
+        if events is not None:
+            events = _check_run_tables(run_imgs, events, "events")
+        if confounds is not None:
+            confounds = _check_run_tables(run_imgs, confounds, "confounds")
+
+        # For each run fit the model and keep only the regression results.
+        nistats_flm.labels_, nistats_flm.results_, self.design_matrices_ = ([], [], [])
+        n_runs = len(run_imgs)
+        t0 = time.time()
+        for run_idx, run_img in enumerate(run_imgs):
```

Comment:
Is it important to do these in series, or are they independent? If they're independent, we may be able to rewrite this to run just one, and use a `MapNode` to iterate.

Reply:
map node sounds good

Reply:
The workflow already uses boldfile as an iterfield in a map node. So the iteration is already done intelligently. the remlfit function (just like the nistats.fit function) has the ability to take a list of images though. I can remove that if you wish in the interest of clarity and brevity.

Reply:
Okay. Let me have another read through before we refactor.